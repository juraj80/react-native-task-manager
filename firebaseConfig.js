import firebase from "firebase/compat/app";
import "firebase/compat/auth";
import "firebase/compat/firestore";
//import { getAuth, signInWithPopup } from "firebase/auth";

const firebaseConfig = {
  apiKey: "AIzaSyBw0NeHk41_45kQpsJeY7dtCzh81Nbzb6Y",
  authDomain: "reacttodoapp-865aa.firebaseapp.com",
  projectId: "reacttodoapp-865aa",
  storageBucket: "reacttodoapp-865aa.appspot.com",
  messagingSenderId: "910067550525",
  appId: "1:910067550525:web:333b134b57ed3843ab5d54",
  measurementId: "G-77GHN72RWN",
};

// Initialize Firebase
if (!firebase.apps.length) {
  firebase.initializeApp(firebaseConfig);
  firebase
    .firestore()
    .settings({ experimentalForceLongPolling: true, merge: true }); //add this..
} else {
  firebase.app();
}

const auth = firebase.auth();
//const auth = getAuth();

export { auth, firebase };
