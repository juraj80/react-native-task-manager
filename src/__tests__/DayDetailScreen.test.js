import React from "react";
import renderer from "react-test-renderer";
import DayDetail from "../DayDetailScreen";
import { firebase } from "../../firebaseConfig";
import {
  render,
  fireEvent,
  screen,
  waitFor,
} from "@testing-library/react-native";

jest.mock("../../firebaseConfig", () => ({
  firebase: {
    firestore: jest.fn().mockReturnValue({
      collection: jest.fn().mockReturnValue({
        onSnapshot: jest.fn().mockImplementation((cb) => {
          cb({
            forEach: jest.fn().mockImplementation((docCb) => {
              docCb({
                id: "some-id",
                data: jest.fn().mockReturnValue({
                  heading: "task heading",
                  text: "task text",
                  completed: false,
                  subtasks: [],
                  dueDateAt: {
                    toDate: jest.fn().mockReturnValue(new Date()),
                  },
                  reminderAt: {
                    toDate: jest.fn().mockReturnValue(new Date()),
                  },
                }),
              });
            }),
          });
        }),
        doc: jest.fn().mockReturnValue({
          delete: jest.fn().mockReturnValue(Promise.resolve()),
        }),
      }),
    }),
  },
}));

jest.mock("@react-navigation/native", () => {
  return {
    ...jest.requireActual("@react-navigation/native"),
    useNavigation: jest.fn(() => ({})),
  };
});

describe("DayDetail", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <DayDetail
          route={{
            params: {
              toISOString: jest
                .fn()
                .mockReturnValue("2022-01-01T12:00:00.000Z"),
            },
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call the firestore collection and onSnapshot method", () => {
    renderer.create(
      <DayDetail
        route={{
          params: {
            toISOString: jest.fn().mockReturnValue("2022-01-01T12:00:00.000Z"),
          },
        }}
      />
    );
    expect(firebase.firestore).toHaveBeenCalledWith();
    expect(firebase.firestore().collection).toHaveBeenCalledWith("tasks");
    expect(firebase.firestore().collection().onSnapshot).toHaveBeenCalledWith(
      expect.any(Function)
    );
  });
});
