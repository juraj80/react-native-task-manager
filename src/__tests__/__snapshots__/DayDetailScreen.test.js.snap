// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayDetail should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(10, 204, 3,1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 20,
        "paddingTop": 60,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "left": "0%",
            "position": "absolute",
            "top": "8%",
          }
        }
      >
        <View
          testID="back-navigation-view"
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
            testID="back-navigation-button"
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                }
              }
              testID="back-navigation-text"
            >
              Back
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "IndieFlower-Regular",
            "fontSize": 34,
            "fontWeight": "bold",
          }
        }
      >
        Tasks
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        2022-01-01
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 15,
          "padding": 0,
        }
      }
    >
      <RCTScrollView
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "height": "100%",
          }
        }
        viewabilityConfigCallbackPairs={Array []}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;
